controls/radvirtualgrid/mvvm.md

```diff
@@ -0,0 +1,327 @@
+	        public MyDataProvider(IEnumerable source) : base(source)
+	        {
+	            this.selectedIndexes = new List<int>();
+	        }
+	
+	        /// <summary>
+	        /// At this state the SelectedIndexes are not cleared yet and can be persisted.
+	        /// </summary>
+	        /// <param name="e"></param>
+	        protected override void SortDescriptorPrepared(SortedEventArgs e)
+	        {
+	            var selectedIndexes = this.ParentGrid.SelectedIndexes.ToList();
+	            if (selectedIndexes.Count > 0)
+	            {
+	                this.PersistSelection(selectedIndexes);
+	            }
+	            base.SortDescriptorPrepared(e);
+	        }
+	
+	        /// <summary>
+	        /// Update the selection after the sorting is processed.
+	        /// </summary>
+	        protected override void OnSortingCompleted()
+	        {
+	            base.OnSortingCompleted();
+	            this.UpdateSelection(this.selectedIndexes);
+	        }
+	
+	        /// <summary>
+	        /// Hide the primary key property
+	        /// </summary>
+	        public override IList<ItemPropertyInfo> ItemProperties
+	        {
+	            get
+	            {
+	                return base.ItemProperties.Skip(1).ToList();
+	            }
+	        }
+	
+	        protected override void OnHeaderValueNeeded(HeaderValueEventArgs e)
+	        {
+	            if (e.HeaderOrientation == VirtualGridOrientation.Horizontal)
+	            {
+	                int propertyIndex = this.GetPropertyIndex("Name");
+	
+	                if (e.Index == propertyIndex)
+	                {
+	                    e.Value = "Club Name";
+	                }
+	                else
+	                {
+	                    base.OnHeaderValueNeeded(e);
+	                }
+	            }
+	        }
+	
+	        protected override void OnEditorNeeded(EditorNeededEventArgs args)
+	        {
+	            int propertyIndex = this.GetPropertyIndex("Established");
+	            if (args.ColumnIndex == propertyIndex)
+	            {
+	                var editor = new RadDateTimePicker();
+	                var item = (this.Source.GetItemAt(args.RowIndex) as Club);
+	
+	                args.Editor = editor;
+	                editor.SelectedValue = item.Established;
+	                args.EditorProperty = RadDateTimePicker.SelectedValueProperty;
+	            }
+	            else
+	            {
+	                base.OnEditorNeeded(args);
+	            }
+	            
+	        }
+	
+	        protected override bool IsColumnReadOnly(int columnIndex)
+	        {
+	            int propertyIndex = this.GetPropertyIndex("Name");
+	            if (columnIndex == propertyIndex)
+	            {
+	                return true;
+	            }
+	
+	            return base.IsColumnReadOnly(columnIndex);
+	        }
+	
+	        /// <summary>
+	        /// Gets the index of a given property
+	        /// </summary>
+	        /// <param name="propertyName">The property name</param>
```

Comment:
the param and returns information is not required here. The client can see this in the public API of the parent class.